configure.ac

```diff
@@ -189,6 +189,11 @@ AC_ARG_WITH(mongoc,
                                          Link against the system supplied or the built-in mongo-c-driver library. (default: auto)]
               ,,with_mongoc="auto")
 
+AC_ARG_ENABLE(legacy-mongodb-options,
+              [  --enable-legacy-mongodb-options=[yes/no]
+                                         Support libmongo-client non-URI MongoDB options. (default: yes)]
+              ,,enable_legacy_mongodb_options="yes")
+
 AC_ARG_WITH(jsonc,
```

Comment:
I am not sure we need this configure option. I understand that we might want to document what portions of the code are related to the compatibility layer, but I don't think we want to make this user controllable.

More options more confusions, and more combinations we need to handle.